pkg/syncer/sync.go

```diff
@@ -145,20 +160,75 @@ func (s *Syncer) SyncPendingCharts(names ...string) error {
 		defer os.RemoveAll(outDir)
 
 		hasDeps := len(ch.Dependencies) > 0
-		tgz, err := chart.ChangeReferences(outDir, ch.TgzPath, ch.Name, ch.Version, s.source, s.target, hasDeps)
+
+		workDir, err := ioutil.TempDir("", "charts-syncer")
+		if err != nil {
+			return errors.Trace(err)
+		}
+		defer os.RemoveAll(workDir)
+		chartPath, err := chart.ChangeReferences(workDir, ch.TgzPath, ch.Name, ch.Version, s.source, s.target)
 		if err != nil {
 			klog.Errorf("unable to process %q chart: %+v", id, err)
 			errs = multierror.Append(errs, errors.Trace(err))
 			continue
 		}
 
+		// Update deps
+		if hasDeps {
+			if err := chart.ChangeDependenciesFile(chartPath, ch.Name, s.source, s.target); err != nil {
+				return errors.Trace(err)
+			}
+			switch s.target.GetRepo().GetKind() {
+			case api.Kind_HELM:
+			case api.Kind_CHARTMUSEUM:
+			case api.Kind_HARBOR:
+				if err := helmcli.UpdateDependencies(chartPath); err != nil {
+					return errors.Trace(err)
+				}
+			case api.Kind_OCI:
+				// Build deps manually for OCI as helm does not support it yet
+				if err := os.RemoveAll(path.Join(chartPath, "charts")); err != nil {
+					return errors.Trace(err)
+				}
+				// Re-create empty charts folder
+				err := os.Mkdir(path.Join(chartPath, "charts"), 0755)
+				if err != nil {
+					return errors.Trace(err)
+				}
+				lock, err := chart.GetChartLock(chartPath, ch.Name)
+				if err != nil {
+					return errors.Trace(err)
+				}
+				// Dependencies found
+				if lock != nil {
+					for _, dep := range lock.Dependencies {
+						depTgz := path.Join(chartPath, "charts", fmt.Sprintf("%s-%s.tgz", dep.Name, dep.Version))
+						s.cli.dst.Fetch(depTgz, dep.Name, dep.Version)
+					}
+				}
```

Comment:
please encapsulate this task into a function, it deserves proper unit testing in an isolated way and the code will look simpler

Reply:
done!